code/iaas/api-logic/src/main/java/io/cattle/platform/iaas/api/auth/github/ProjectResourceManager.java

```diff
@@ -156,6 +193,68 @@ private String getExternalId(String externalId, String externalIdType, String jw
         throw new ClientVisibleException(ResponseCodes.BAD_REQUEST, "UnrecognizedScope", "Scope " + externalIdType + " is invalid", null);
     }
 
+    @Override
+    protected Object deleteInternal(String type, String id, final Object obj, ApiRequest apiRequest) {
+        if (!(obj instanceof Account)) {
+            return new Object();
+        }
+        String token = getTokenFromRequest(apiRequest);
+        objectProcessManager.executeStandardProcess(StandardProcess.DEACTIVATE, obj,
+                ProcessUtils.chainInData(new HashMap<String, Object>(), AccountConstants.ACCOUNT_DEACTIVATE, AccountConstants.ACCOUNT_REMOVE));
+        Account deletedAccount = (Account) objectManager.reload(obj);
+        GithubUtils.ReverseMappings reverseMappings = githubUtils.validateAndFetchReverseMappings(token);
+        AccountPolicy policy = (AccountPolicy) ApiContext.getContext().getPolicy();
+        Account modifiedAccount = transformProject(deletedAccount, reverseMappings);
+        policy.grantObjectAccess(modifiedAccount);
+        return ImmutableList.of(modifiedAccount);
```

Comment:
Avoid trickling guava everywhere.  There's not a great advantage of this over the standard Arrays.asList(...).  I know it's not immutable, but that's not that big of deal.  I'll be honest, I'm not a big fan of Guava, so if you use it it better provide some real value.  The reason it's on the classpath is just for listenablefutures, in general I prefer apache utils.